### PR TITLE
Allowed Snowstorms to happen - added snow wet and glare effects ( and fixed test_weather )

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Enabled snowstorms - wet and glare effects for snow"```

#### Purpose of change
Snowstorms were coded but never happened due to the condition for them to appear.
Snow had no weather effects.
Test_weather was outputting numbers seperated by a comma, to a csv file, messing up the separation, it also didnt output all weather types, it also used a random seed each call to get_weather.

#### Describe the solution
I've enabled a rare-ish condition for snowstorms to appear.
Snow and Snowstorms add wet effects ( less than rain )
Snow and snowstorms add snow_glare effect.
Gave a constant seed to test_weather, made the separators a semi-colon.
Added weather_type ( cloudy, sunny ) and wind speed and direction to the weather test output.

#### Describe alternatives you've considered
N/A
#### Additional context
N/A

### CODE REVIEW

**Comment by KITbutler** [comment]
I'm not familiar with the weather code but to me it looks like you are applying the snow_glare **while** the snow(storm) is occurring. I hope that statement in itself illustrates the mistake.

**Comment by davidpwbrown** [comment]
Suggesting that during a snow storm, there would be so little light there would be little glare...
Yes.
Unfortunately the game has no way to track how much snow is on the ground...
I will remove from snow storm and keep on snowing, snow storm has reduced visibility as it is.

**Comment by KITbutler** [comment]
If it isn't a thing already try adding glare to winter sunrise+sunset (or just *all* sunny periods) since [this](https://c8.alamy.com/comp/S1T8YG/hazardous-winter-driving-in-rural-ontario-canada-with-snow-blowing-S1T8YG.jpg) is what I would call "decent" visibility when facing the winter sun.

File: src/weather.cpp [hunk 1]
```diff
@@ -68,6 +69,32 @@ void weather_effect::glare()
         }
     }
 }
+/**
+ * Snow Glare.
+ * Causes glare effect to player's eyes if they are not wearing applicable eye protection.
+ */
+void weather_effect::snow_glare()
```
Comment (ZhilkinSerg) [inline src/weather.cpp]: Is this a copy of `weather_effect::glare()`? Maybe you should move shared code to some function and use it in both methods?

Comment (davidpwbrown) [reply src/weather.cpp]: No, it doesn't work, I'll keep trying.

Comment (davidpwbrown) [reply src/weather.cpp]: ok got it now

**Comment by davidpwbrown** [comment]
Sorry to add feature while being reviewed, but it's little.
Just added a windspeed condition before snowstorm is created.
and added a snowstorm trigger to the soundpack environments, if a soundpack wishes to make use of it.

File: src/sounds.cpp [hunk 2]
```diff
@@ -606,6 +607,8 @@ void sfx::do_ambient()
             case WEATHER_SUNNY:
             case WEATHER_CLOUDY:
             case WEATHER_SNOWSTORM:
+                play_ambient_variant_sound( "environment", "WEATHER_SNOWSTORM", heard_volume, 20,
```
Comment (ZhilkinSerg) [inline src/sounds.cpp]: I believe `WEATHER_SNOWSTORM` would now play for sunny,cloudy and snowstorm weather. That's not your totally your fault though - previously `WEATHER_SNOW` would now play for sunny, cloudy and snow weather.

**Pull request merged**